Plugins/Flow.Launcher.Plugin.PluginsManager/PluginsManager.cs

```diff
@@ -0,0 +1,198 @@
+
+        internal PluginsManager(PluginInitContext context)
+        {
+            pluginsManifest = new PluginsManifest();
+            Context = context;
+        }
+        internal void InstallOrUpdate(UserPlugin plugin)
+        {
+            if (PluginExists(plugin.ID)) 
+            {
+                Context.API.ShowMsg("Plugin already installed");
+                return;
+            }
+
+            var message = string.Format(Context.API.GetTranslation("plugin_pluginsmanager_install_prompt"), 
+                                                                        Environment.NewLine, Environment.NewLine, 
+                                                                        plugin.Name, plugin.Author);
+
+            if(MessageBox.Show(message, Context.API.GetTranslation("plugin_pluginsmanager_install_title"), MessageBoxButton.YesNo) == MessageBoxResult.No)
+                return;
+
+            var filePath = Path.Combine(DataLocation.PluginsDirectory, $"{plugin.Name}{plugin.ID}.zip");
+            
+            try
+            {
+                Context.API.ShowMsg(Context.API.GetTranslation("plugin_pluginsmanager_downloading_plugin"),
+                                    Context.API.GetTranslation("plugin_pluginsmanager_please_wait"));
+
+                Utilities.Download(plugin.UrlDownload, filePath);
+
+                Context.API.ShowMsg(Context.API.GetTranslation("plugin_pluginsmanager_downloading_plugin"),
+                                    Context.API.GetTranslation("plugin_pluginsmanager_download_success"));
+            }
+            catch (Exception e)
+            {
+                Context.API.ShowMsg(Context.API.GetTranslation("plugin_pluginsmanager_downloading_plugin"),
+                                Context.API.GetTranslation("plugin_pluginsmanager_download_success"));
+
+                Log.Exception("PluginsManager", "An error occured while downloading plugin", e, "PluginDownload");
+            }
+
+            Application.Current.Dispatcher.Invoke(() => Install(plugin, filePath));
+        }
+
+        internal void Update()
+        {
+            throw new NotImplementedException();
+        }
+
+        internal bool PluginExists(string id)
+        {
+            return Context.API.GetAllPlugins().Any(x => x.Metadata.ID == id);
+        }
+
+        internal void PluginsManifestSiteOpen()
+        {
+            //Open from context menu https://git.vcmq.workers.dev/Flow-Launcher/Flow.Launcher.PluginsManifest
+            throw new NotImplementedException();
+        }
+
+        internal List<Result> Search(List<Result> results, string searchName)
+        {
+            if (string.IsNullOrEmpty(searchName))
+                return results;
+
+            return results
+                    .Where(x => StringMatcher.FuzzySearch(searchName, x.Title).IsSearchPrecisionScoreMet())
+                    .Select(x => x)
+                    .ToList();
```

Comment:
Why not use the score to order it?
Like 
```c#
.Where(x=>{
x.Score = StringMatcher.FuzzySearch(searchName,x.Title));
return x.Score>0;
})
```

Reply:
score > 0 will return a lot of things that have a score, that's why we are using IsSearchPrecisionScoreMet(), it matches the precision level that the user selected, default is 50, so only results with a score equal or higher will return.

Reply:
Well, you are not right here. The score property will make every raw score lower than search precision become 0, so no worry for more results. The one you suggested that can be lower than search precision is called rawScore.

Reply:
yeah you are right score.

I dont see a benefit of doing x.score > 0 as oppose to just calling the function which has the logic embedded without having to hardcode any numbers

Reply:
I mean by changing x.score to the score from the fuzzy search can make the result list sort since the view model will sort it by its score.

Reply:
can you give me a code snippet please, not sure what you mean

Reply:
I mean by setting the score of each Result can help the result list to sort it based on the score.
Check out this https://github.com/Flow-Launcher/Flow.Launcher/blob/920668db0042460b8d08a86e89f44e2385159012/Flow.Launcher/ViewModel/ResultsViewModel.cs#L156-L201

Although I don't believe this part of codes are clear enough, but it do provides the ability to sorts the results to display on the results list view based on their score.

Reply:
Ohh gotcha, I forgot to set the score for each of the result.

Reply:
sorted. good catch, thank you